internal/config/config.go

```diff
@@ -1065,6 +1065,18 @@ func buildAggregator(name string, tbl *ast.Table) (*models.AggregatorConfig, err
 		}
 	}
 
+	if node, ok := tbl.Fields["allow_historical"]; ok {
```

Comment:
am I correctly inferring that this is internal to telegraf? would this work for all plugins?

Reply:
It would work for all aggregator type plugins. Which means it extends all existing aggregator plugins with added functionality, but because the default value is false, it does not have adverse affects to existing deployments.